File: tests/adapters/kotlin/goldens/literals/max_tokens_10.kt
/**
 * Kotlin module for testing literal optimization.
 */

package com.example.literals

// Short string literal (should be preserved)
const val SHORT_MESSAGE = "Hello, World!"

// Long string literal (candidate for trimming)
const val LONG_MESSAGE = """This is an extremely long message that cont…""" // literal string (−63 tokens)

// Multi-line raw string with embedded expressions
val TEMPLATE_WITH_DATA = """
User Information:
- Name: ${g…""" // literal string (−58 tokens)

data class DataContainer(
    // Small array (should be preserved)
    val tags: List<String>,
    
    // Large array (candidate for trimming)
    val items: List<String>,
    
    // Small object (should be preserved)
    val metadata: Map<String, Any>,
    
    // Large object (candidate for trimming)
    val configuration: Map<String, Any>
)

class LiteralDataManager {
    // Class properties with various literal types
    private val smallConfig = mapOf(
        "debug" to true,
        "…" to "…"
    ) // literal object (−5 tokens)
    
    private val largeConfig = mapOf(
        "…" to "…"
    ) // literal object (−334 tokens)
    
    private val supportedLanguages: List<String>
    private val allowedExtensions: Set<String>
    
    init {
        // Array with many elements (trimming candidate)
        supportedLanguages = listOf(
            "english",
            "spanish",
            "…"
        ) // literal array (−89 tokens)
        
        // Set with many elements
        allowedExtensions = setOf(
            ".kt",
            ".kts",
            "…"
        ) // literal set (−47 tokens)
    }
    
    fun processData(): DataContainer {
        // Function with various literal data
        val smallArray = listOf("one", "two", "…")
        
        val largeArray = listOf(
            "item_001",
            "…"
        ) // literal array (−146 tokens)
        
        val nestedData = mapOf(
            "…" to "…"
        ) // literal object (−205 tokens)
        
        return DataContainer(
            tags = smallArray,
            items = largeArray,
            metadata = mapOf("type" to "test", "…" to "…"), // literal object (−2 tokens)
            configuration = nestedData
        )
    }
    
    fun getLongQuery(): String {
        // Very long SQL-like query string
        return """
            SELECT 
                use…""" /* literal string (−181 tokens) */.trimIndent()
    }
}

// Module-level constants with different sizes
val SMALL_CONSTANTS = mapOf(
    "API_VERSION" to "v1",
    "…" to "…"
) // literal object (−2 tokens)

val LARGE_CONSTANTS = mapOf(
    "…" to "…"
) // literal object (−556 tokens)

// Helper functions that use literal data
fun getUserName(): String = "John Doe"
fun getUserEmail(): String = "john.doe@example.com"
fun getAccountStatus(): String = "active"
fun getPermissions(): List<String> = listOf("read", "write", "…")
fun getLastLogin(): String = "2024-01-15T1…" // literal string (−5 tokens)
fun getProfileCompleteness(): Int = 85

File: tests/adapters/javascript/goldens/literals/max_tokens_10.js
/**
 * JavaScript module for testing literal optimization.
 */

// Short string literal (should be preserved)
const SHORT_MESSAGE = "Hello, World!";

// Long string literal (candidate for trimming)
const LONG_MESSAGE = `This is an extremely long message that conta…`; // literal string (−65 tokens)

// Multi-line template literal with embedded expressions
const TEMPLATE_WITH_DATA = `
User Information:
- Name: ${getUserName()}…`; // literal string (−57 tokens)

export class LiteralDataManager {
    // Class properties with various literal types
    #smallConfig = {
        debug: true,
        version: "1.0.0"
    };

    #largeConfig = {
        database: {
            host: "localhost",
            // … (5 more, −75 tokens)
        },
        // … (3 more, −145 tokens)
    };

    constructor() {
        // Array with many elements (trimming candidate)
        this.supportedLanguages = [
            "english",
            "…",
        ]; // literal array (−89 tokens)

        // Set with many elements
        this.allowedExtensions = new Set([
            ".js",
            "…",
        ]); // literal array (−56 tokens)
    }

    processData() {
        // Function with various literal data
        const smallArray = ["one", "two", "three"];

        const largeArray = [
            "item_001",
            "…",
        ]; // literal array (−145 tokens)

        const nestedData = {
            level1: {
                level2: {
                    level3: {
                        data: [
                            { id: 1, name: "First", active: true },
                            "…",
                        ] // literal array (−61 tokens),
                        // … (1 more, −41 tokens)
                    }
                }
            }
        };

        return {
            tags: smallArray,
            // … (3 more, −22 tokens)
        };
    }

    getLongQuery() {
        // Very long SQL-like query string
        return `
            SELECT
                use…`; // literal string (−182 tokens)
    }
}

// Module-level constants with different sizes
export const SMALL_CONSTANTS = {
    API_VERSION: "v1",
    DEFAULT_LIMIT: 50
};

export const LARGE_CONSTANTS = {
    HTTP_STATUS_CODES: {
        CONTINUE: 100,
        // … (40 more, −241 tokens)
    },
    // … (1 more, −122 tokens)
};

// Helper functions that use literal data
function getUserName() { return "John Doe"; }
function getUserEmail() { return "john.doe@example.com"; }
function getAccountStatus() { return "active"; }
function getPermissions() { return ["read", "write", "admin"]; }
function getLastLogin() { return "2024-01-15T1…"; /* literal string (−7 tokens) */ }
function getProfileCompleteness() { return 85; }
